receiver/statsdreceiver/internal/protocol/statsd_parser.go

```diff
@@ -305,6 +305,14 @@ func (p *StatsDParser) Aggregate(line string, addr net.Addr) error {
 		p.instrumentsByAddress[addrKey] = instrument
 	}
 
+	for _, parsedMetric := range parsedMetrics {
+		p.aggregateMetric(instrument, parsedMetric)
+	}
+
+	return nil
+}
+
+func (p *StatsDParser) aggregateMetric(instrument *instruments, parsedMetric statsDMetric) {
```

Comment:
I think this could be simpler if `asFloat64Slice []float64` was on `statsDMetric` struct itself, so we wouldn't need to copy over the metadata.

Reply:
This was my first initial thought but that will absolutely allocate on the heap, even if return is only shortly lived.